nilearn/connectome/connectivity_matrices.py

```diff
@@ -466,20 +455,59 @@ def fit(self, X, y=None):
                              "provided arrays of dimensions "
                              "{0}".format(str(subjects_dims)))
 
-        n_subjects = [s.shape[1] for s in X]
-        if len(set(n_subjects)) > 1:
+        features_dims = [s.shape[1] for s in X]
+        if len(set(features_dims)) > 1:
             raise ValueError("All subjects must have the same number of "
                              "features.\nYou provided: "
-                             "{0}".format(str(n_subjects)))
+                             "{0}".format(str(features_dims)))
 
-        if self.kind == 'tangent':
-            covariances = [self.cov_estimator_.fit(x).covariance_ for x in X]
+    def fit(self, X, y=None):
+        """Fit the covariance estimator to the given time series for each
+        subject.
+
+        Parameters
+        ----------
+        X : list of numpy.ndarray, shape for each (n_samples, n_features)
+            The input subjects time series. The number of samples may differ
+            from one subject to another.
+
+        Returns
+        -------
+        self : ConnectivityMatrix instance
+            The object itself. Useful for chaining operations.
+        """
+        self._check_input(X)
+        self.cov_estimator_ = clone(self.cov_estimator)
+
+        covariances = [self.cov_estimator_.fit(x).covariance_ for x in X]
+        if self.kind == 'tangent_geometric':
             self.mean_ = _geometric_mean(covariances, max_iter=30, tol=1e-7)
+        else:
+            self.mean_ = np.mean(covariances, axis=0)
+            # Fight numerical instabilities: make symmetric
+            self.mean_ = self.mean_ + self.mean_.T
```

Comment:
If I understood correctly, this also applies for all kinds except tangent_geometric ?